javalib/src/main/scala/java/lang/System.scala

```diff
@@ -100,14 +99,79 @@ private object SystemProperties {
   import System.{lineSeparator, getenv}
 
   private val systemProperties0 = loadProperties()
-  val systemProperties = {
+  private val systemProperties = {
     Platform.setOSProps { (key: CString, value: CString) =>
       systemProperties0.setProperty(fromCString(key), fromCString(value))
       ()
     }
     systemProperties0
   }
```

Comment:
Maybe move inside to make a stack var?
```suggestion
  private val systemProperties = {
    val systemProperties0 = loadProperties()
    Platform.setOSProps { (key: CString, value: CString) =>
      systemProperties0.setProperty(fromCString(key), fromCString(value))
      ()
    }
    systemProperties0
  }
```

Reply:
I tried it didn't work. The `CFuncPtr` can access only static memory and I get an error at compile time.

Reply:
Interesting that it can't close over a local variable. Does that make sense to you as a limitation?

Reply:
Yes, it is a known limitation of the C ffi

Reply:
Can you have a Scala function that closes over the `val` and then is converted to a C function?

Reply:
You can't directly, but there are workarounds for that.
Usually C libraries allow to pass a pointer as data bucket in callbacks. You can get the pointer to a Scala object (after adding it as root in the GC) and pass it to the c library.
When the callback is executed you cast that pointer back to the scala object, and so you can close to any dynamic data you want.
An example is PortData in SNUnit: https://github.com/lolgab/snunit/blob/dfaea2d338d66c0688324e26c1022abedb83a30b/snunit-async-cats-effect/src/snunit/CEAsyncServerBuilder.scala#L75